notebook/services/contents/manager.py

```diff
@@ -317,22 +317,29 @@ def increment_filename(self, filename, path='', insert=''):
             The name of a file, including extension
         path : unicode
             The API path of the target's directory
+        insert: unicode
+            The characters to insert after the base filename
 
         Returns
         -------
         name : unicode
             A filename that is unique, based on the input filename.
         """
+        # Extract the full suffix from the filename (e.g. .tar.gz)
         path = path.strip('/')
-        basename, ext = os.path.splitext(filename)
+        dirname, basename = os.path.split(filename)
+        name, dot, ext = basename.partition('.')
+        basename = os.path.join(dirname, name)
+        suffix = dot + ext
+
         for i in itertools.count():
             if i:
                 insert_i = '{}{}'.format(insert, i)
             else:
                 insert_i = ''
-            name = u'{basename}{insert}{ext}'.format(basename=basename,
-                insert=insert_i, ext=ext)
-            if not self.exists(u'{}/{}'.format(path, name)):
+            name = u'{basename}{insert}{suffix}'.format(basename=basename,
+                insert=insert_i, suffix=suffix)
+            if not self.exists(os.path.join(path, name)):
```

Comment:
Ah, hang on. This function works with an API path, which is always `/` separated, rather than a platform-dependent native path. So using `os.path.split` and `os.path.join` is wrong here.

Reply:
Updated.  Looking again, `filename` should never have a directory in it.